rules/type-declaration/src/TypeInferer/PropertyTypeInferer/ConstructorPropertyTypeInferer.php

```diff
@@ -61,7 +62,13 @@ public function inferProperty(Property $property): Type
 
         // A. infer from type declaration of parameter
         if ($param->type !== null) {
-            return $this->resolveFromParamType($param, $classMethod, $propertyName);
+            $type = $this->resolveFromParamType($param, $classMethod, $propertyName);
+
+            if ($param->variadic) {
+                return new ArrayType(new ConstantIntegerType(0), $type);
```

Comment:
Or should it be ?

```suggestion
                return new ArrayType(new IntegerType(), $type);
```

Reply:
The `resolveFromParamType()` method should be fixed instead, as it's output is incorrect

Reply:
Like this https://github.com/rectorphp/rector/pull/5359/commits/e8bb282c30dbf3a58d35bb09ef60a5add5ded3a8 ?

Reply:
Super :)